pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -474,18 +473,24 @@ func newAccumulator(base Extent) (*accumulator, error) {
 	}, nil
 }
 
-func toExtent(ctx context.Context, req MetricsQueryRequest, res Response, queryTime time.Time, samplesProcessed uint64) (Extent, error) {
+func toExtent(ctx context.Context, req MetricsQueryRequest, res Response, queryTime time.Time, perStepStats []stats.StepStat) (Extent, error) {
 	marshalled, err := types.MarshalAny(res)
 	if err != nil {
 		return Extent{}, err
 	}
+
+	extentPerStepStats := make([]StepStat, len(perStepStats))
+	for i, stat := range perStepStats {
+		extentPerStepStats[i] = StepStat(stat)
+	}
```

Comment:
do you have any numbers on how much this would increase the cache size? should we try to delta-encode the per-step stats for example?

Reply:
This PR will not increase the cache size; it changes the method of calculating cachedStats. However, they have been stored in a cache for ~ a month now after [this](https://github.com/grafana/mimir/pull/11164) was merged and released.
Speaking about numbers, it's protobuf encoded in cache, so I didn't do any real math around using delta-encoding.

Reply:
I kept an eye on the Query results cache hit ratio hit rate when was testing it in dev cell and in ops after the rollout, (AFAIK it was released in ops on 04-24) and I didn't notice any sudden hit rate drops. Is there anything else we can check?